app/controllers/products_controller.rb

```diff
@@ -1,29 +1,58 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only:[:show, :edit, :update, :destroy]
+ before_action :set_product, only:[:show, :edit, :update, :destroy]
   
   def index
+    @products = Product.all
   end
-  
-  def show
-  end
-  
+
+  def show; end
+
   def dashboard
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
   end
 
   def create
+    begin
```

Comment:
you don't need to use `begin` keyword when you are doing the rescue for the whole implementation